securedrop_client/gui/widgets.py

```diff
@@ -19,181 +19,553 @@
+        color: #fff;
+    }
+    '''
+
+    def __init__(self):
+        super().__init__()
+
+        # Set css id
+        self.setObjectName('activity_status_bar')
+
+        # Set styles
+        self.setStyleSheet(self.css)
+
+        # Remove grip image at bottom right-hand corner
+        self.setSizeGripEnabled(False)
+
+    def update_message(self, message: str, duration: int):
         """
-        Display a status message to the user. Optionally, supply a duration
-        (in milliseconds), the default will continuously show the message.
+        Display a status message to the user.
         """
         self.showMessage(message, duration)
 
-    def hide_refresh_icon(self):
+
+class ErrorStatusBar(QWidget):
+    """
+    A pop-up status bar for displaying messages about application errors to the user. Messages will
+    be displayed for a given duration or until the message is cleared or updated with a new message.
+    """
+
+    css = '''
+    #error_vertical_bar {
+        background-color: #f22b5d;
+    }
+    #error_icon {
+        background-color: qlineargradient(
+            x1: 0,
+            y1: 0,
+            x2: 0,
+            y2: 1,
+            stop: 0 #fff,
+            stop: 0.2 #fff,
+            stop: 1 #fff
+        );
+    }
+    #error_status_bar {
+        background-color: qlineargradient(
+            x1: 0,
+            y1: 0,
+            x2: 0,
+            y2: 1,
+            stop: 0 #fff,
+            stop: 0.2 #fff,
+            stop: 1 #fff
+        );
+        font-weight: bold;
+        color: #f22b5d;
+    }
+    '''
+
+    def __init__(self):
+        super().__init__()
+
+        # Set styles
+        self.setStyleSheet(self.css)
+
+        # Set layout
+        layout = QHBoxLayout(self)
+        self.setLayout(layout)
+
+        # Remove margins and spacing
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+
+        # Error vertical bar
+        self.vertical_bar = QWidget()
+        self.vertical_bar.setObjectName('error_vertical_bar')
+        self.vertical_bar.setFixedWidth(10)
+
+        # Error icon
+        self.label = SvgLabel('error_icon.svg', svg_size=QSize(32, 32))
+        self.label.setObjectName('error_icon')
+        self.label.setFixedWidth(42)
+
+        # Error status bar
+        self.status_bar = QStatusBar()
+        self.status_bar.setObjectName('error_status_bar')
+        self.status_bar.setSizeGripEnabled(False)
+
+        # Add widgets to layout
+        layout.addWidget(self.vertical_bar)
+        layout.addWidget(self.label)
+        layout.addWidget(self.status_bar)
+
+        # Hide until a message needs to be displayed
+        self.vertical_bar.hide()
+        self.label.hide()
+        self.status_bar.hide()
+
+        # Only show errors for a set duration
+        self.status_timer = QTimer()
+        self.status_timer.timeout.connect(self._on_status_timeout)
+
+    def _hide(self):
+        self.vertical_bar.hide()
```

Comment:
We should make sure each of these objects has the `parent` value set so that hiding the parent hides all the children. This would make the show/hide methods unnecessary. I don't know if this is possible in this context, and if not it might make sense for us to rewrite it work like that. I think in general in our code we should be using the parent/child relationships more, and this is one place it seems like it would be beneficial.